chapter08/spg/exercise2.pl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+my $file_handles = {};
+
+open my $input_file, '<', 'datafile.txt' or die "Could not open datafile.txt: $!";
+
+while ( <$input_file> ) {
+  my $line = readline $input_file;
+
+  next if ( !defined $line || $line !~ /^([^:]+):/ );
+
+  my ( $name, undef ) = split ( ':', $line );

```

Comment:
You can omit the second `undef` entry - undef as an assignment binding is useful when you want to skip an element, say if you wanted `my ( $name, undef, $favourite_colour )`